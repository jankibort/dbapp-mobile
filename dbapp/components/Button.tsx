import React, { FC } from 'react';
import {
  NativeSyntheticEvent,
  NativeTouchEvent,
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  ViewStyle,
} from 'react-native';

type Props = {
  title: string | JSX.Element;
  onPress: (ev: NativeSyntheticEvent<NativeTouchEvent>) => void;
  buttonStyle?: ViewStyle | ViewStyle[];
  labelStyle?: StyleProp<TextStyle> | StyleProp<TextStyle>[];
};

export const Button: FC<Props> = ({
  title,
  onPress,
  buttonStyle,
  labelStyle,
  ...rest
}) => {
  return (
    <Pressable
      onPress={onPress}
      style={[buttonStyle, styles.buttonShape]}
      {...rest}
    >
      <Text style={[labelStyle]}>{title}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  buttonShape: {
    alignSelf: 'flex-start',
    textAlignVertical: 'center',
    alignItems: 'center',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderRadius: 6,
    justifyContent: 'space-around',
    alignContent: 'center',
    minHeight: 26,
  },
});
