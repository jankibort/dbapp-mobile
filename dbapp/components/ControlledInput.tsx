import React, { FC } from 'react';
import { StyleSheet, TextInput } from 'react-native';
import { COLORS } from '../constants';

type Props = {
  onChange: ((text: string) => void) | undefined;
  value: string | number;
  type?: 'text' | 'password';
};

export const ControlledInput: FC<Props> = ({
  onChange,
  value,
  type = 'text',
  ...restProps
}) => {
  return (
    <TextInput
      {...restProps}
      value={value}
      onChangeText={onChange}
      style={styles.input}
      secureTextEntry={type === 'password' && true}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    backgroundColor: COLORS.LIGHT,
    borderColor: COLORS.PRIMARY,
    height: 36,
    marginVertical: 12,
    borderWidth: 1,
    padding: 10,
    borderRadius: 6,
  },
});
