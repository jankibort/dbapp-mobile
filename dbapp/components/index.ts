export { Button } from './Button';
export { ControlledInput } from './ControlledInput';
