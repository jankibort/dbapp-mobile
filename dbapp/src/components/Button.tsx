import React, { FC } from 'react';
import {
  NativeSyntheticEvent,
  NativeTouchEvent,
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  ViewStyle,
} from 'react-native';

export type TButtonProps = {
  title: string | JSX.Element;
  onPress: (ev: NativeSyntheticEvent<NativeTouchEvent>) => void;
  buttonStyle?: ViewStyle | ViewStyle[];
  labelStyle?: StyleProp<TextStyle> | StyleProp<TextStyle>[];
  testID?: string;
};

export const Button: FC<TButtonProps> = ({
  title,
  onPress,
  buttonStyle,
  labelStyle,
  testID,
  ...rest
}) => {
  return (
    <Pressable
      onPress={onPress}
      style={[buttonStyle, styles.buttonShape]}
      testID={testID}
      {...rest}
    >
      <Text style={[labelStyle]}>{title}</Text>
    </Pressable>
  );
};

export const styles = StyleSheet.create({
  buttonShape: {
    alignSelf: 'flex-start',
    textAlignVertical: 'center',
    alignItems: 'center',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderRadius: 6,
    justifyContent: 'space-around',
    alignContent: 'center',
    minHeight: 26,
  },
});
