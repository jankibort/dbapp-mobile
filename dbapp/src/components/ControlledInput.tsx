import React, { FC } from 'react';
import { StyleSheet, TextInput } from 'react-native';
import { COLORS } from '../constant';

type Props = {
  onChange: ((text: string) => void) | undefined;
  value?: string;
  type?: 'text' | 'password';
  disabled?: boolean;
};

export const ControlledInput: FC<Props> = ({
  onChange,
  value,
  type = 'text',
  disabled,
  ...restProps
}) => {
  return (
    <TextInput
      {...restProps}
      value={value}
      onChangeText={onChange}
      style={[styles.input, disabled && styles.disabled]}
      secureTextEntry={type === 'password' && true}
      editable={!disabled}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    backgroundColor: COLORS.LIGHT,
    borderColor: COLORS.PRIMARY,
    height: 36,
    marginVertical: 12,
    borderWidth: 1,
    padding: 10,
    borderRadius: 6,
  },
  disabled: {
    backgroundColor: COLORS.DISABLED,
  },
});
