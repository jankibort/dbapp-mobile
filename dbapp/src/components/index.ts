export * from './Button';
export { ControlledInput } from './ControlledInput';
