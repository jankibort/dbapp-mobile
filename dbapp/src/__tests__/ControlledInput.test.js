import React from 'react';
import renderer from 'react-test-renderer';
import { ControlledInput } from '../components/ControlledInput';

test('renders correctly', () => {
  const tree = renderer.create(<ControlledInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
