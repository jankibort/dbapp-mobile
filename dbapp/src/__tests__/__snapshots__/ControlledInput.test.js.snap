// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  rejectResponderTermination={true}
  secureTextEntry={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "#317cf5",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 36,
        "marginVertical": 12,
        "padding": 10,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
